### ISSUE TITLE
How to preserve /goss/docker_output.log on failure?

### ISSUE THREAD

**Issue by eifinger** [aelsabbahy/goss#386]
Hi,
I use dgoss in my CI pipeline to test my docker images.
I have a http test defined and if it fails I would like to know what exactly went wrong.

The way I understand the dgoss script it cleans up after a failure and thus deletes /goss/docker_output.log

What would be the best way to get the output of my docker container on failure and print it to my build log?

Thank you!

**Comment by tholok97** [comment]
Wondering about this as well.

**Comment by tholok97** [comment]
Got dgoss to print to the build log by adding a `docker logs "$id"` to the EXIT trap before cleanup like this:

```
# Main
tmp_dir=$(mktemp -d /tmp/tmp.XXXXXXXXXX)
chmod 777 "$tmp_dir"
trap 'ret=$?;docker logs "$id";cleanup;exit $ret' EXIT
```

Could someone please explain how this is normally acieved? Having the container log in the build log seems like a very legitimate thing to want, so I'm surprised it's not there by default.

**Comment by aelsabbahy** [comment]
merged/relesed fix in #519
